plugins/sdex.go

```diff
@@ -813,3 +813,19 @@ func (sdex *SDEX) GetLatestTradeCursor() (interface{}, error) {
 
 	return records[0].PT, nil
 }
+
+// GetOrderBook gets the SDEX order book
+func (sdex *SDEX) GetOrderBook(pair *model.TradingPair) (orderBook horizon.OrderBookSummary, e error) {
```

Comment:
this should conform to the API method described in the previous comments:

```GetOrderBook(pair *model.TradingPair, maxCount int32) (*model.OrderBook, error)```

this would require transforming orders from horizon's format to the `model.OrderBook` format
I can give you sample code for this if needed, I have something.

Reply:
@Reidmcc the sample code below converts a single `manageOffer` to a `model.Order`.
You'd need to do something similar, converting from `horizon.OrderBookSummary` to `model.OrderBook` (which mostly contains a list of `model.Order`). Hope this is useful.

```golang
func manageOffer2Order(mob *build.ManageOfferBuilder, baseAsset horizon.Asset, quoteAsset horizon.Asset) (*model.Order, error) {
	orderAction := model.OrderActionSell
	price := model.NumberFromFloat(float64(mob.MO.Price.N)/float64(mob.MO.Price.D), utils.SdexPrecision)
	volume := model.NumberFromFloat(float64(mob.MO.Amount)/math.Pow(10, 7), utils.SdexPrecision)
	isBuy, e := assetsEqual(quoteAsset, mob.MO.Selling)
	if e != nil {
		return nil, fmt.Errorf("could not compare assets, error: %s", e)
	}
	if isBuy {
		orderAction = model.OrderActionBuy
		// TODO need to test price and volume conversions correctly
		// volume calculation needs to happen first since it uses the non-inverted price when multiplying
		volume = model.NumberFromFloat(volume.AsFloat()*price.AsFloat(), utils.SdexPrecision)
		price = model.InvertNumber(price)
	}

	return &model.Order{
		Pair: &model.TradingPair{
			Base:  model.FromHorizonAsset(baseAsset),
			Quote: model.FromHorizonAsset(quoteAsset),
		},
		OrderAction: orderAction,
		OrderType:   model.OrderTypeLimit,
		Price:       price,
		Volume:      volume,
		Timestamp:   model.MakeTimestamp(time.Now().UnixNano() / int64(time.Millisecond)),
	}, nil
}
```